packages/meditrak-app/app/assessment/actions/submitSurvey/createUpsertEntityObjects.js

```diff
@@ -23,13 +24,18 @@ const buildEntity = async (state, database, fields, entityId, answers) => {
     }
   });
 
+  const isUpdate = database.findOne('Entity', entityId);
+  if (isUpdate) {
+    return entity;
+  }
+
   const getSelectedCountry = () => database.getCountry(state.country.selectedCountryId);
   if (!entity.countryCode) {
     entity.countryCode = getSelectedCountry().code;
   }
   if (!entity.parent) {
```

Comment:
Hey @rohan-bes and @chris-pollard I'm just looking through this to update datatrak. I'm wondering if this should be `entity.parent_id` and same on line 21?

Reply:
@tcaiger actually the way that realm database records Entity records is that the parent column records an Entity itself, which then has the parent_id, which is extracted for syncing purposes.